diff --git a/plugins/a-fluidsynth.lv2/a-fluidsynth.ttl.in b/plugins/a-fluidsynth.lv2/a-fluidsynth.ttl.in
index be6a82a..a4c7a0a 100644
--- a/plugins/a-fluidsynth.lv2/a-fluidsynth.ttl.in
+++ b/plugins/a-fluidsynth.lv2/a-fluidsynth.ttl.in
@@ -3,6 +3,7 @@
 @prefix foaf:  <http://xmlns.com/foaf/0.1/> .
 @prefix lv2:   <http://lv2plug.in/ns/lv2core#> .
 @prefix midi:  <http://lv2plug.in/ns/ext/midi#> .
+@prefix mod:   <http://moddevices.com/ns/mod#> .
 @prefix patch: <http://lv2plug.in/ns/ext/patch#> .
 @prefix pprop: <http://lv2plug.in/ns/ext/port-props#> .
 @prefix rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@@ -19,6 +20,7 @@
 
 <urn:distrho:a-fluidsynth:sf2file>
   a lv2:Parameter ;
+  mod:fileTypes "sf2" ;
   rdfs:label "SF2 File" ;
   rdfs:range atom:Path .
 
diff --git a/plugins/a-fluidsynth.lv2/fluidsynth/config.h b/plugins/a-fluidsynth.lv2/fluidsynth/config.h
index df62444..aaefb78 100644
--- a/plugins/a-fluidsynth.lv2/fluidsynth/config.h
+++ b/plugins/a-fluidsynth.lv2/fluidsynth/config.h
@@ -150,10 +150,10 @@
 /* #undef LADSPA */
 
 /* libsndfile has ogg vorbis support */
-/* #undef LIBSNDFILE_HASVORBIS */
+#define LIBSNDFILE_HASVORBIS 1
 
 /* Define to enable libsndfile support */
-/* #undef LIBSNDFILE_SUPPORT */
+#define LIBSNDFILE_SUPPORT 1
 
 /* Define to enable MidiShare driver */
 /* #undef MIDISHARE_SUPPORT */
@@ -201,11 +201,7 @@
 #define VERSION "2.0.6"
 
 /* Define to do all DSP in single floating point precision */
-#ifdef __arm__
-#  define WITH_FLOAT
-#else
-#  undef WITH_FLOAT
-#endif
+#define WITH_FLOAT 1
 
 /* Define to profile the DSP code */
 /* #undef WITH_PROFILING */
diff --git a/plugins/a-fluidsynth.lv2/fluidsynth/fluidsynth/fluidsynth.h b/plugins/a-fluidsynth.lv2/fluidsynth/fluidsynth/fluidsynth.h
index 8852ea2..fcef55a 100644
--- a/plugins/a-fluidsynth.lv2/fluidsynth/fluidsynth/fluidsynth.h
+++ b/plugins/a-fluidsynth.lv2/fluidsynth/fluidsynth/fluidsynth.h
@@ -3,7 +3,7 @@
 
 #include <stdio.h>
 
-#if defined(COMPILER_MSVC)
+#if defined(COMPILER_MSVC) || defined(FLUIDSYNTH_NOT_A_DLL)
 #  define FLUIDSYNTH_API
 #else
 #  define FLUIDSYNTH_API  __attribute__ ((visibility ("hidden")))
diff --git a/plugins/a-fluidsynth.lv2/a-fluidsynth.cc b/plugins/a-fluidsynth.lv2/a-fluidsynth.cc
index fcf7154..e43de59 100644
--- a/plugins/a-fluidsynth.lv2/a-fluidsynth.cc
+++ b/plugins/a-fluidsynth.lv2/a-fluidsynth.cc
@@ -723,7 +723,11 @@ save (LV2_Handle                instance,
 {
 	AFluidSynth* self = (AFluidSynth*)instance;
 
-	if (strlen (self->current_sf2_file_path) == 0) {
+	const char* file_path = self->queue_reinit
+	                      ? self->queue_sf2_file_path
+	                      : self->current_sf2_file_path;
+
+	if (strlen (file_path) == 0) {
 		return LV2_STATE_ERR_NO_PROPERTY;
 	}
 
@@ -742,7 +746,7 @@ save (LV2_Handle                instance,
 		return LV2_STATE_ERR_NO_FEATURE;
 	}
 
-	char* apath = map_path->abstract_path (map_path->handle, self->current_sf2_file_path);
+	char* apath = map_path->abstract_path (map_path->handle, file_path);
 	store (handle, self->afs_sf2file,
 			apath, strlen (apath) + 1,
 			self->atom_Path, LV2_STATE_IS_POD);
